Source/Shared/arcana/threading/task.h

```diff
@@ -352,7 +352,7 @@ namespace arcana
 
         for (task<void, ErrorT>& task : tasks)
         {
-            task.then(inline_scheduler, cancellation::none(), [data, result](const basic_expected<void, ErrorT>& exp) mutable noexcept {
```

Comment:
I had to do this for things to compile, but I don't quite understand the implications of removing it. @ryantrem Any thoughts?

Reply:
In the early days of the Arcana task system, only `std::error_code` was supported (it was not templated on the error type), so this could just be a holdover from that. If we have unit tests that cover `when_all` for both `std::error_code` and `std::exception_ptr` then I suspect we're ok.

Reply:
I added one for when_all with exceptions. My concern is whether there are any implications of removing `noexcept`.

Reply:
I fixed the issue by checking the type of ErrorT for now which is already being done elsewhere.

For context, `noexcept` can have performance implications and thus we should use it as much as possible. See https://en.cppreference.com/w/cpp/language/noexcept_spec.